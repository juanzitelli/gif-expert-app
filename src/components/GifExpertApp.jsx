import React, { useState } from 'react'
import AddCategory from './AddCategory'
import GifGrid from './GifGrid'

const GifExpertApp = ({defaultCategories = []}) => {
	const [categories, setCategories] = useState(defaultCategories)

	return (
		<>
			<div className="appHeader animate__animated animate__backInDown">
				<h2>GifExpertApp <span role="img" aria-label="emoji">ðŸ˜Ž</span> </h2>

				<div>
					<i>Search</i>
					<AddCategory setCategories={setCategories} />
				</div>
			</div>
			<div>
				{
					categories.map(category => <GifGrid key={Math.random()} category={category} />)
					//CAMBIAR ESTE MATH RANDOM POR ID
				}
			</div>

		</>
	)
}

export default GifExpertApp
