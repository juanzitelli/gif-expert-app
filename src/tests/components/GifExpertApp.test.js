import React from "react";
import "@testing-library/jest-dom";
// import GifGrid from "../../components/GifGrid";
import { shallow } from "enzyme";
import GifExpertApp from "../../components/GifExpertApp";
// import { useFetchGifs } from "../../hooks/useFetchGifs";
import "@testing-library/jest-dom";

describe("Pruebas en GifExpertApp", () => {
  test("Debe renderizarse correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  }); 

  test("Debe mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
