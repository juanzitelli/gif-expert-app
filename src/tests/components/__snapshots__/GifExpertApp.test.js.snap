// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <div
    className="appHeader animate__animated animate__backInDown"
  >
    <h2>
      GifExpertApp 
      <span
        aria-label="emoji"
        role="img"
      >
        ðŸ˜Ž
      </span>
       
    </h2>
    <div>
      <i>
        Search
      </i>
      <AddCategory
        setCategories={[Function]}
      />
    </div>
  </div>
  <div>
    <GifGrid
      category="One Punch"
      key="0.8886936985370169"
    />
    <GifGrid
      category="Dragon Ball"
      key="0.3933647464984589"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Debe renderizarse correctamente 1`] = `
<Fragment>
  <div
    className="appHeader animate__animated animate__backInDown"
  >
    <h2>
      GifExpertApp 
      <span
        aria-label="emoji"
        role="img"
      >
        ðŸ˜Ž
      </span>
       
    </h2>
    <div>
      <i>
        Search
      </i>
      <AddCategory
        setCategories={[Function]}
      />
    </div>
  </div>
  <div />
</Fragment>
`;
